plugin/evm/header/block_gas_cost.go

```diff
@@ -0,0 +1,61 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package header
+
+import (
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/commontype"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/plugin/evm/blockgascost"
+)
+
+// BlockGasCost calculates the required block gas cost based on the parent
+// header and the timestamp of the new block.
+func BlockGasCost(
```

Comment:
I notice this has a different type than coreth. Is the intention to have 2 packages for custom header? (sort of makes sense)

Reply:
I think we can merge it by passing coreth fee configs as a commontype.FeeConfig. These will be refactored in several times (with upcoming cherry picked commits), as long as I see in the final form we have very similar fee configs in `ap3, ap4` packages. So we can just define them as `ap4Configs commontype.FeeConfig` etc. (wishful thinking)